src/overlays/actors/ovl_Obj_Boyo/z_obj_boyo.c

```diff
@@ -19,22 +26,183 @@ const ActorInit Obj_Boyo_InitVars = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+static ObjBoyoUnkStruct sCollisionHandlers[] = {
+    { ACTOR_PLAYER, ObjBoyo_PushPlayer },
+    { ACTOR_EN_KAIZOKU, ObjBoyo_PushPirate },
+    { ACTOR_EN_BOM, ObjBoyo_ExplodeBomb },
+};
+
+void ObjBoyo_Init(Actor* thisx, PlayState* play) {
+    ObjBoyo* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Collider_InitCylinder(play, &this->collider);
+    Collider_SetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->animatedMaterial = Lib_SegmentedToVirtual(&object_boyo_Matanimheader_000E88);
+}
+
+void ObjBoyo_Destroy(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    ObjBoyo* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void ObjBoyo_PushPlayer(ObjBoyo* this, Actor* actor) {
+    Player* player = (Player*)actor;
+
+    player->pushedSpeed = 30.0f;
+    player->pushedYaw = this->actor.yawTowardsPlayer;
+}
+
+void ObjBoyo_PushPirate(ObjBoyo* this, Actor* actor) {
+    EnKaizoku* kaizoku = (EnKaizoku*)actor;
+
+    kaizoku->unk_2F0 = 30.0f;
+    kaizoku->unk_2F4 = Actor_WorldYawTowardActor(&this->actor, &kaizoku->picto.actor);
+}
+
+void ObjBoyo_ExplodeBomb(ObjBoyo* this, Actor* actor) {
+    EnBom* bom = (EnBom*)actor;
+
+    bom->timer = 0;
+}
+
+/*
+ * Find the Actor we collided with.
+ */
+Actor* ObjBoyo_FindCollidedActor(ObjBoyo* this, PlayState* play, s32* index) {
+    Actor* collidedActor;
+    ObjBoyoUnkStruct* iter;
+    s32 i;
+
+    if (this->collider.base.ocFlags2 & OC2_HIT_PLAYER) {
+        *index = 0;
+        return &GET_PLAYER(play)->actor;
+    }
+    iter = sCollisionHandlers + 1;
+    if (this->collider.base.ocFlags1 & OC1_HIT) {
+        collidedActor = this->collider.base.oc;
+        for (i = 1; i != 3; i++) {
+            if (iter->id == collidedActor->id) {
+                *index = i;
+                return collidedActor;
+            }
+            iter++;
+        }
+    }
+    return NULL;
+}
+
+void ObjBoyo_Update(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    ObjBoyo* this = THIS;
+    Actor* collidedActor;
+    s32 index;
+
+    collidedActor = ObjBoyo_FindCollidedActor(this, play, &index);
+
+    if (collidedActor != NULL) {
+        sCollisionHandlers[index].colHandler(this, collidedActor);
+        // TODO: find out what all of these are.
+
+        this->unk194 = 0x64;
+        this->unk196 = 3;
+        this->unk198 = 0.01f;
+        this->unk19C = 0.03f;
+        this->unk1A4 = 0x3F40;
+        this->unk1A6 = 0x7D0;
+        this->unk1A8 = 0;
+        this->unk1A0 = 0.03f;
+        this->unk1AA = 0x2DF7;
+        this->unk1AC = 0x258;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boyo/ObjBoyo_Destroy.s")
+    if (this->unk194 > 0) { /* compute new scaling */
```

Comment:
```suggestion
    // compute new scaling
    if (this->unk194 > 0) {
```
Comments in function are usually styled like this